component/src/governance/check.rs

```diff
@@ -291,4 +261,81 @@ pub mod stateful {
 
         Ok(())
     }
+
+    pub async fn proposal_withdraw(
+        state: &State,
+        proposal_withdraw: &ProposalWithdraw,
+    ) -> Result<()> {
+        // Any voteable proposal can be withdrawn
+        proposal_voteable(state, proposal_withdraw.proposal).await?;
```

Comment:
Any voteable proposal can be withdrawn, because if the transaction balances, that means you are spending the corresponding proposal NFT, which you could only have if you submitted the proposal. No further checks here are necessary because of this.